src/CommunityToolkit/Xamarin.CommunityToolkit/Effects/Shadow/PlatformShadowEffect.uwp.cs

```diff
@@ -0,0 +1,200 @@
+﻿using System;
+using System.ComponentModel;
+using System.Linq;
+using System.Numerics;
+using Windows.UI.Composition;
+using Windows.UI.Xaml;
+using Windows.UI.Xaml.Controls;
+using Windows.UI.Xaml.Hosting;
+using Windows.UI.Xaml.Shapes;
+using Xamarin.CommunityToolkit.Effects;
+using Xamarin.CommunityToolkit.UWP.Effects;
+using Xamarin.Forms;
+using Xamarin.Forms.Platform.UWP;
+using Grid = Xamarin.Forms.Grid;
+using Image = Windows.UI.Xaml.Controls.Image;
+
+[assembly: ExportEffect(typeof(PlatformShadowEffect), nameof(ShadowEffect))]
+
+namespace Xamarin.CommunityToolkit.UWP.Effects
+{
+	public class PlatformShadowEffect : PlatformEffect
+	{
+		enum ShadowEffectState
+		{
+			Initialized,
+			PanelCreated,
+			Attached
+		}
+
+		const float defaultRadius = 10f;
+
+		const float defaultOpacity = 1f;
+
+		ShadowEffectState state;
+
+		SpriteVisual? spriteVisual;
+
+		Layout<View>? shadowPanel;
+
+		DropShadow? shadow;
+
+		FrameworkElement View => Control ?? Container;
+
+		protected override void OnElementPropertyChanged(PropertyChangedEventArgs args)
+		{
+			base.OnElementPropertyChanged(args);
+
+			switch (args.PropertyName)
+			{
+				case nameof(ShadowEffect.ColorPropertyName):
+				case nameof(ShadowEffect.OpacityPropertyName):
+				case nameof(ShadowEffect.RadiusPropertyName):
+				case nameof(ShadowEffect.OffsetXPropertyName):
+				case nameof(ShadowEffect.OffsetYPropertyName):
+					UpdateShadow();
+					break;
+			}
+		}
+
+		protected override void OnAttached()
+		{
+			if (Element is not View elementView)
+				return;
+
+			switch (state)
+			{
+				case ShadowEffectState.Attached:
+					return;
+
+				case ShadowEffectState.Initialized:
+					shadowPanel = new StackLayout()
+					{
+						Children = { new Grid() }
+					};
+					state = ShadowEffectState.PanelCreated;
+
+					MoveElementTo(elementView, shadowPanel);
+					return;
+
+				case ShadowEffectState.PanelCreated:
+					AppendShadow();
+					state = ShadowEffectState.Attached;
+					return;
+
+				default:
+					throw new ArgumentOutOfRangeException();
```

Comment:
```suggestion
				case ShadowEffectState.Initialized:
					shadowPanel = new StackLayout()
					{
						Children = { new Grid() }
					};
					
					state = ShadowEffectState.PanelCreated;
					MoveElementTo(elementView, shadowPanel);
					break;
				case ShadowEffectState.PanelCreated:
					AppendShadow();
					state = ShadowEffectState.Attached;
					break;
				default:
					break;
```

Reply:
Maybe without default: break; ?

Reply:
Why not?) Do you expect that new values appear in this enum?

Reply:
Definitely nope) I mean there is no code after switch)

Reply:
ah, gotcha) so yeah, both break and return can work here)